iroh/src/endpoint.rs

```diff
@@ -1581,25 +1547,17 @@ impl Connection {
     /// connection.
     ///
     /// [`PublicKey`]: iroh_base::PublicKey
-    // TODO: Would be nice if this could be infallible.
-    pub fn remote_node_id(&self) -> Result<NodeId> {
-        let data = self.peer_identity();
-        match data {
-            None => bail!("no peer certificate found"),
-            Some(data) => match data.downcast::<Vec<rustls::pki_types::CertificateDer>>() {
-                Ok(certs) => {
-                    if certs.len() != 1 {
-                        bail!(
-                            "expected a single peer certificate, but {} found",
-                            certs.len()
-                        );
-                    }
-                    let cert = tls::certificate::parse(&certs[0])?;
-                    Ok(cert.peer_id())
-                }
-                Err(_) => bail!("invalid peer certificate"),
-            },
-        }
+    pub fn remote_node_id(&self) -> NodeId {
+        let data = self
+            .inner
+            .peer_identity()
+            .expect("required by iroh crypto setup");
+        let certs = data
+            .downcast::<Vec<rustls::pki_types::CertificateDer>>()
+            .expect("iroh crypto setup");
+        debug_assert_eq!(certs.len(), 1);
+        let cert = tls::certificate::parse(&certs[0]).expect("TODO: is this guaranteed by now?");
+        cert.peer_id()
```

Comment:
I'd like better expect messages here, too. "required by iroh crypto setup" doesn't suffice IMO.
The message should mention the place that the `peer_identity` field is populated.
I tried figuring this out, but yeah - it's probably not that easy (flows through quinn, right?).